src/components/ha-icon-picker.ts

```diff
@@ -87,48 +130,57 @@ export class HaIconPicker extends LitElement {
     `;
   }
 
-  private async _openedChanged(ev: PolymerChangedEvent<boolean>) {
-    this._opened = ev.detail.value;
-    if (this._opened && !iconLoaded) {
-      const iconList = await import("../../build/mdi/iconList.json");
+  // Filter can take a significant chunk of frame (up to 3-5 ms)
+  private _filterIcons = memoizeOne(
+    (filter: string, iconItems: IconItem[] = ICONS) => {
```

Comment:
It's a filter so it should return `IconItem[]` for `RankedIcon[]`

Reply:
Not sure what change you are suggesting.  It returns `RankedIcon[]` unless the filter string is empty, in which case it returns `IconItem[]` just to avoid the expense of creating a new array.  And in the filtered case, it just avoids the unnecessary copies of `parts` and `keywords`.

Reply:
@piitaya do you want to get some beta testing for this?  I can make final tweaks but I don't know what you are requesting from your comment.